sdk/python/feast/errors.py

```diff
@@ -133,7 +133,7 @@ def __init__(self, name, project=None):
 
 
 class RequestDataNotFoundInEntityDfException(FeastObjectNotFoundException):
```

Comment:
Instead of using an optional second parameter, you can update the implementation of `from_error_detail` and replace this line:
```py
                instance = class_reference(message)
```
with:
```py
                instance = class_reference.__new__(class_reference)
```
Sorry for not suggesting this before.